### PR TITLE
Protocol analysis/enum own file def

### PR DESCRIPTION
**Pull request by Jaguar0625**
1. move all supporting types to {plugin}_types.cats files
2. split cosignature and aggregate schemas
3. move MosaicNonce to types files

### CODE REVIEW

**Comment by Jaguar0625** [comment]
other comments
1. lock_hash_algorithm.cats renamed to lock_secret_types.cats since it was only named enum file
2. transaction files only have transaction types all other types were moved to plugin types files

File: schemas/mosaic/mosaic_definition.cats [hunk 1]
```diff
@@ -1,35 +1,6 @@
...
-	# mosaic supports custom restrictions configured by mosaic owner
-	restrictable = 0x04
-
-# available mosaic property ids
-enum MosaicPropertyId : uint8
-	# mosaic duration
-	duration = 0x02
-
-# mosaic property compose of an id and a value
-struct MosaicProperty
```
Comment (dgarcia360) [inline schemas/mosaic/mosaic_definition.cats]: I will probably not move MosaicProperty from this file, only the enums.

File: schemas/multisig/modify_multisig_account.cats [hunk 2]
```diff
@@ -1,21 +1,6 @@
...
 
-# cosignatory modification type
-enum CosignatoryModificationType : uint8
-	# add cosignatory
-	add = 0x00
-
-	# remove cosignatory
-	del = 0x01
-
-# binary layout for a cosignatory modification
```
Comment (dgarcia360) [inline schemas/multisig/modify_multisig_account.cats]: I would not move CosignatoryModification from this file, only the enums.

File: schemas/restriction_account/account_address_restriction.cats [hunk 3]
```diff
@@ -1,11 +1,6 @@
-import "restriction_account/account_restriction_types.cats"
+import "restriction_account/restriction_account_types.cats"
 import "transaction.cats"
 
-# account address restriction modification
-struct AccountAddressRestrictionModification
```
Comment (dgarcia360) [inline schemas/restriction_account/account_address_restriction.cats]: I would not move CosignatoryModification from this file, only the enums.

File: schemas/restriction_account/account_mosaic_restriction.cats [hunk 4]
```diff
@@ -1,11 +1,6 @@
-import "restriction_account/account_restriction_types.cats"
+import "restriction_account/restriction_account_types.cats"
 import "transaction.cats"
 
-# account mosaic restriction modification
-struct AccountMosaicRestrictionModification
```
Comment (dgarcia360) [inline schemas/restriction_account/account_mosaic_restriction.cats]: I would not move AccountMosaicRestrictionModification from this file as it is not an enumeration.

**Comment by Jaguar0625** [comment]
Discussed with @dgarcia360 offline - and decided following is ok provided consistent across all files

> move dependent types into types files so that tx files just have txes then other types are easier to find 

Thanks for reviewing!

**Pull request closed without merging**